src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -141,35 +141,35 @@ public static void ValidateArgumentCount(MethodBase method, ExpressionType nodeK
             }
         }
 
-        public static Expression ValidateOneArgument(MethodBase method, ExpressionType nodeKind, Expression arg, ParameterInfo pi)
+        public static Expression ValidateOneArgument(MethodBase method, ExpressionType nodeKind, Expression arguments, ParameterInfo pi)
```

Comment:
`arg` here comes from `arguments` arguments to public members, and hence should be the argument name in an exception that goes back to such methods, either by changing the name or by not using `nameof` in this case.

It does though mean that at this point we have an `arguments` for a single argument.